packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -239,20 +250,35 @@ export class ArtifactCreator {
     // variables are captured in its closure scope. TypeScript compiler
     // currently has no mechanism to detect this and will happily tell you
     // your code is correct when it isn't.
-    await page.evaluate(async () => {
-      const modelBecomesReady = (self as any).modelLoaded ?
-          Promise.resolve() :
-          new Promise((resolve) => {
-            // const timeout = setTimeout(reject, 60000);
+    const evaluateError =
+        await page.evaluate(async (useMaxTime, maxTimeInSec) => {
+          const modelBecomesReady = new Promise((resolve, reject) => {
+            let timeout: NodeJS.Timeout;
+            if (useMaxTime) {
+              timeout = setTimeout(reject, maxTimeInSec * 1000);
+            }
 
             self.addEventListener('model-ready', () => {
-              // clearTimeout(timeout);
+              if (useMaxTime) {
+                clearTimeout(timeout);
+              }
               resolve();
             }, {once: true});
           });
 
-      await modelBecomesReady;
-    });
+          try {
+            await modelBecomesReady;
+            return null;
+          } catch (error) {
+            return `Stop capturing screenshot after ${maxTimeInSec} seconds`;
```

Comment:
I think you can put this error message into your reject(). Then you could display whatever error message you get, because it's possible you got an error that isn't the timeout.